src/api.js

```diff
@@ -0,0 +1,34 @@
+const BASE_URL = import.meta.env.VITE_BASE_URL;
+const API_KEY = import.meta.env.VITE_API_KEY;
+
+function fetchWithTimeout(url, options, timeout = 5000) {
+  return Promise.race([
+    fetch(url, options),
+    new Promise((_, reject) =>
+      setTimeout(
+        () => reject(new Error("요청 시간이 초과되었습니다.")),
+        timeout
+      )
+    ),
+  ]);
+}
+
+export async function fetchMovies(category, page = 1) {
+  const url = `${BASE_URL}/${category}?language=ko-KR&page=${page}`;
+  const options = {
+    method: "GET",
+    headers: {
+      accept: "application/json",
+      Authorization: `Bearer ${API_KEY}`,
+    },
+  };
+
+  try {
+    const response = await fetchWithTimeout(url, options, 10000);
```

Comment:
numeric separator를 사용해주시면 조금 더 잘 읽힐거 같네용 ㅎㅎ 
```suggestion
    const response = await fetchWithTimeout(url, options, 10_000);
```